model/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProvider.java

```diff
@@ -116,13 +119,30 @@ static EventEntity convertEvent(Event event) {
         eventEntity.setIpAddress(event.getIpAddress());
         eventEntity.setError(event.getError());
         try {
-            eventEntity.setDetailsJson(mapper.writeValueAsString(event.getDetails()));
+            if (maxDetailLength > 0 && event.getDetails() != null) {
+                Map<String, String> result = new HashMap<>(event.getDetails());
+                result.entrySet().forEach(t -> t.setValue(trimToMaxLength(t.getValue())));
+
+                eventEntity.setDetailsJson(mapper.writeValueAsString(result));
+            } else {
+                eventEntity.setDetailsJson(mapper.writeValueAsString(event.getDetails()));
+            }
         } catch (IOException ex) {
             logger.error("Failed to write log details", ex);
         }
         return eventEntity;
     }
 
+    private String trimToMaxLength(String detail) {
+        if (detail != null && detail.length() > maxDetailLength) {
+            String result = detail.substring(0, maxDetailLength).concat("...");
```

Comment:
It should take `...` into account in the max-length, so the string with `...` should be max maxDetailLength long